packages/clerk-js/src/ui/components/UserProfile/EnterpriseAccountsSection.tsx

```diff
@@ -15,58 +32,115 @@ export const EnterpriseAccountsSection = () => {
-                    colorScheme='body'
-                  >
-                    {providerName}
-                  </Text>
-                  <Text
-                    truncate
-                    as='span'
-                    colorScheme='secondary'
-                  >
-                    {label ? `• ${label}` : ''}
-                  </Text>
-                  {error && (
-                    <Badge
-                      colorScheme='danger'
-                      localizationKey={localizationKeys('badge__requiresAction')}
-                    />
-                  )}
-                </Flex>
-              </Box>
-            </ProfileSection.Item>
-          );
-        })}
+        {activeEnterpriseAccounts.map(account => (
+          <EnterpriseAccount
+            key={account.id}
+            account={account}
+          />
+        ))}
       </ProfileSection.ItemList>
     </ProfileSection.Root>
   );
 };
+
+const EnterpriseAccount = ({ account }: { account: EnterpriseAccountResource }) => {
+  const label = account.emailAddress;
+  const providerName = getEnterpriseAccountProviderName(account);
+  const error = account.verification?.error?.longMessage;
+
+  return (
+    <ProfileSection.Item
+      id='enterpriseAccounts'
+      sx={t => ({
+        gap: t.space.$2,
+        justifyContent: 'start',
+      })}
+      key={account.id}
+    >
+      <EnterpriseAccountProviderIcon account={account} />
+      <Box sx={{ whiteSpace: 'nowrap', overflow: 'hidden' }}>
+        <Flex
+          gap={2}
+          center
+        >
+          <Text
+            truncate
+            colorScheme='body'
+          >
+            {providerName}
+          </Text>
+          <Text
+            truncate
+            as='span'
+            colorScheme='secondary'
+          >
+            {label ? `• ${label}` : ''}
+          </Text>
+          {error && (
+            <Badge
+              colorScheme='danger'
+              localizationKey={localizationKeys('badge__requiresAction')}
+            />
+          )}
+        </Flex>
+      </Box>
+    </ProfileSection.Item>
+  );
+};
+
+const EnterpriseAccountProviderIcon = ({ account }: { account: EnterpriseAccountResource }) => {
+  const { provider, enterpriseConnection } = account;
+  const providerName = getEnterpriseAccountProviderName(account) ?? provider;
+
+  if (isOAuthCustomProvider(provider)) {
+    return enterpriseConnection?.logoPublicUrl ? (
+      <Image
+        elementDescriptor={[descriptors.providerIcon]}
+        elementId={descriptors.providerIcon.setId(provider)}
+        alt={providerName}
+        src={enterpriseConnection.logoPublicUrl}
+        sx={theme => ({ width: theme.sizes.$4 })}
+      />
+    ) : (
+      <ProviderInitialIcon
+        id={provider}
+        value={providerName}
+        aria-label={`${providerName}'s icon`}
+      />
+    );
+  }
+
+  const providerWithoutPrefix = provider.replace('oauth_', '').trim() as OAuthProvider;
```

Comment:
We talked about this a bit with @Nikpolik . It's a bit weird that we're stripping the `oauth_` here, but it's certainly not the first time we've done it. Sometimes `provider` has that `oauth_` prefix, and sometimes it doesn't.

It would be good for us to figure out what we want `provider` to mean, and then standardize it across the board. But that can happen in the future, and shouldn't block this PR.

Reply:
It is being used in element descriptors, so we should feel confidant with its usage before this goes out since it'd be a breaking change to remove/modify it.

Reply:
@alexcarpenter Yeah, we need to be mindful of updating `provider` values in the future, since element descriptors rely on it and it could be a breaking change.

On this PR, I'm keeping the behavior as before, where we have the following variations: 
- `enterpriseButtonsProviderIcon__ saml_okta`
- `enterpriseButtonsProviderIcon__ google`
- `enterpriseButtonsProviderIcon__ custom_roblox` (could be any other custom provider name)